pages/api/user/search.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from "../../../lib/endpoints/express";
+import nc from "next-connect";
+import mongoose from "../../../lib/middlewares/mongoose";
+import auth from "../../../lib/middlewares/auth";
+import { isLoggedIn } from "../../../lib/middlewares/user";
+import User from "../../../models/User";
+import IUser from "../../../types/IUser";
+import { send } from "process";
+
+const handler = nc<Request, Response>();
+
+handler.use(mongoose, ...auth, isLoggedIn);
+
+/**
+ * find all of the users that are a partial name match for the given query
+ * string
+ */
+handler.get(async (req, res) => {
+  // get the query string
+  const query = req.query.q as String;
+
+  if (query.length < 2) {
```

Comment:
You never check if `query` exists -- it could be undefined here, which would cause a 500 error. We should check to make sure that query is defined in this statement prior to checking the length.